pkgs/development/libraries/libgudev/default.nix

```
@@ -47,6 +47,7 @@ stdenv.mkDerivation (finalAttrs: {
   ];
 
   doCheck = true;
+  mesonFlags = lib.optional (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) "-Dtests=disabled";
```

```suggestion
  mesonFlags = lib.optional (!finalAttrs.finalPackage.doCheck) "-Dtests=disabled";
```

`finalAttrs.finalPackage.doCheck` contains the result of the `doCheck` logic in make-derivation.nix